cromwell-backend/src/main/scala/cromwell/backend/config/BackendConfiguration.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.backend.config
+
+import com.typesafe.config.ConfigFactory
+
+import scala.collection.JavaConversions._
+
+/**
+  * Defines a backend configuration.
+  * @param name Backend name.
+  * @param initClass Initialization class for the specific backend.
+  */
+case class BackendConfigurationEntry(name: String, initClass: String)
+
+object BackendConfiguration {
+  val config = ConfigFactory.load()
+  val backendCfg = config.getConfig("backend")
+  val defaultBackend = backendCfg.getString("default")
+  val backendProviders = backendCfg.getConfigList("providers")
+  val backendList = backendProviders.map(entry =>
+    BackendConfigurationEntry(entry.getString("name"), entry.getString("initClass"))).toList
+
+  def apply(): BackendConfiguration = new BackendConfiguration(backendList, defaultBackend)
+}
+
+/**
+  * Retrieves backend configuration.
+  * @param backendList List of backend entries in configuration file.
+  * @param defaultBackend Backend name to be used as default.
+  */
+class BackendConfiguration(backendList: List[BackendConfigurationEntry], defaultBackend: String) {
```

Comment:
That's kinda what I'm saying, that I don't see the point of the extra-genericness on these things anymore.

My thoughts are definitely evolving, I think it's sparked by whatever vestiges of java-tinged OOP still remain in my psyche finally fading away, but @mcovarr and I have certainly debated this off and on for a while now. I'm at a point where I know what I *don't* like but haven't really come up with what I *do* like and seem to have fallen back to List as "grab bag of stuff where I don't particularly care about the exact structure". There's also traversable but a) I've always hated how long a name that is and b) it has all of the negatives of Seq (except I don't think it's secretly a List like Seq usually is - which is an annoying source of bugs).

Either way I'm not at a phase where I'm really going to pick on people in reviews, but in this case was just saying that I'm personally trending in the other direction :)

Much like CWL support I don't think this is really a topic we should be polluting this code review with but if you wanted to continue off-thread (in person/whatever) I'm happy to. Or perhaps you don't care and are willing to let old man gentry go back to his mumblings and rumblings ;)